Modules/Filtering/ImageGrid/test/itkBSplineScatteredDataPointSetToImageFilterTest5.cxx

```diff
@@ -115,19 +112,37 @@ itkBSplineScatteredDataPointSetToImageFilterTest5(int argc, char * argv[])
   close.Fill(1);
   filter->SetCloseDimension(close);
 
-
   ITK_TRY_EXPECT_NO_EXCEPTION(filter->Update());
 
-  // Cast the PhiLattice
-  using CastImageFilterType = itk::CastImageFilter<FilterType::PointDataImageType, OutputImageType>;
-  auto caster = CastImageFilterType::New();
-  caster->SetInput(filter->GetPhiLattice());
+  // using OutputPixelType = unsigned char;
+  using OutputPixelType = uint8_t;
+  using OutputImageType = itk::Image<OutputPixelType, ParametricDimension>;
+
+  auto                     phiLatticeImage = filter->GetPhiLattice();
+  OutputImageType::Pointer outputImage = OutputImageType::New();
+  outputImage->CopyInformation(phiLatticeImage);
+  outputImage->SetRegions(phiLatticeImage->GetBufferedRegion());
+  outputImage->Allocate(true);
+
+  // phiLatticeImage contains pixels of type Vector<double, 3>
+  // with minmax values of (-5, 5) and magnitude max of about 5
+  auto myiterator =
+    itk::ImageRegionIterator<FilterType::PointDataImageType>(phiLatticeImage, phiLatticeImage->GetBufferedRegion());
+  auto myOutiterator = itk::ImageRegionIterator<OutputImageType>(outputImage, outputImage->GetBufferedRegion());
+  while (!myiterator.IsAtEnd())
+  {
+    // Convert vector image to magnitude and scale to use range of png values
+    constexpr double max_magnitude_value = 5.0;
```

Comment:
If magnitude is <= 5, this looks good.

Reply:
OK.  I think I'll dynamically compute the value.  The 5 is hard-coded here in an attempt to get the draft in front of you. The next version will dynamically compute max_magnitude value.  

I'll also add similar fixes in other failing tests. It might be a few days before I get them done.

Reply:
I now dynamically compute the max value.